docs/src/userguide/extension_types.rst

```diff
@@ -284,7 +284,7 @@ when it is deleted.::
 
         cdef object cheeses
 
-        def __cinit__(self):
+        def __init__(self):
```

Comment:
Given that there are no arguments, I think `__cinit__` is fine here. It prevents the attribute from being `None` early on.

Reply:
Using `__init__` allows Cython to automatically generate the `__reduce__` method for pickling and copying the object. If the attribute was an extension type, then None could be unsafe (since nonecheck=False by default). But in this case, the code is still memory safe since the attribute is typed as an object,

Reply:
@scoder Do you still want me to change it back? I am not sure if you read my last comment.

Reply:
Looking at the full example I think `cheeses` should be typed as `list`, and the initializer kept as `__cinit__`.

Clearly it's not a completely definitive "this is the right way to write it". But that's probably how I'd write it.

Reply:
Okay, then. Well it looks like the master branch pull ahead of me. I have deleted my change to this example, so merging this PR will not change the current example on master (which has `cheeses` typed as `object`, and keeps `__cinit__`). @da-woods Would you like me to change `cheeses` to be typed as `list`? Doing so will require rebasing.

Reply:
I think we should just merge this now with the examples we all agree on. We don't have to change everything here - it's not like using cinit is wrong.

I'll do that when the tests have all finished